docs/vectorizer/worker.md

```diff
@@ -1,53 +1,136 @@
 
-## Prerequisites
 
-To run vectorizer workers, you need to:
+## Running a vectorizer worker as a CLI tool
 
-* Install:
-  * Container environment: [Docker][docker]
-  * Local environment: [Python3][python3] and [pip][pip]
-  * All environments: A Postgres client like [psql][psql]
-* Create a key for your AI provider:
-  * [OpenAI][openai-key]
-  * [Voyage AI][voyage-key]
+**Prerequisites**: [Python3][python3] and [pip][pip]
 
-## Install and configure vectorizer worker
 
-To be able to run vectorizers in your self-hosted database, use one of the following setups:
+1. **Install [pgai](https://pypi.org/project/pgai/) from PyPI**
+
+   ```shell
+   pip install pgai
+   ```
+
+   The `pgai` command line tool should now be in your `$PATH`.
+   
+1. **Create a .env file**
+
+   [Configure](#setting-api-keys-through-environment-variables-or-env-file) the API keys for your embedding providers by adding them to a `.env` file. For example, if you are using OpenAI, you can add the following:
+
+   ```
+   OPENAI_API_KEY=<your-openai-api-key>
+   ```
+   
+   Alternatively, you can set the API key through an environment variable, but this is less secure.
+
+1. **Run the vectorizer worker**
+
+    After you [define a vectorizer in your database](/docs/vectorizer.md#define-a-vectorizer), you run 
+    a vectorizer worker to generate and update your embeddings:
+    
+    ```shell
+    pgai vectorizer worker -d <db-connection-string>
+    ```
+
+For more configuration options, see [Advanced configuration options](#advanced-configuration-options) below.
+
+## Running a vectorizer worker in your own application 
+
+ **Prerequisites**: [Python3][python3] and [pip][pip]
+
+1. Add the pgai package dependency to your project
+
+   ```shell
+   pip install pgai
+   ```
+   or add `pgai` to the dependencies in your `requirements.txt` file, `pyproject.toml`, or similar configuration file.
+
+2. Add the vectorizer worker to run in the background of your application
+
+   ```python
+   from pgai import Worker
+
+   worker = Worker(db_url=<your-database-connection-string>)
+   task = asyncio.create_task(worker.run())
+   ```
+
+   You can then shutdown the worker gracefully when your application shuts down:
+
+   ```python
+    
+    await worker.request_graceful_shutdown()
+    try:
+        result = await asyncio.wait_for(task, timeout=20)
+        if result is not None:
+            print("Worker shutdown with exception:", result)
+        else:
+            print("Worker shutdown successfully")
+    except asyncio.TimeoutError:
+        print("Worker did not shutdown in time, it was killed")
+   ```
+   
+3. Make sure you add the API keys for your embedding providers to the environment variables when you run 
+  your application. 
+  
+    We recommend using a `.env` file to set the API key and then load the `.env` file using the `load_dotenv` function from the `python-dotenv` package.
```

Comment:
Does the user have to run `load_dotenv()` or does the worker does this anyway? We should probably clarify that here.